nntrainer/tensor/manager.h

```diff
@@ -261,12 +261,15 @@ class Manager {
    * @brief Allocate memory for all the managed tensors
    */
   void allocateTensors() {
-    /// Weights are allocated while initializing
-    if (!weights_initialized)
+    if (!weights_allocated)
       allocateWeights();
-    allocateGradients();
-    allocateInOuts();
-    allocateDerivatives();
+
+    if (!tensors_allocated) {
+      allocateGradients();
+      allocateInOuts();
+      allocateDerivatives();
+      tensors_allocated = true;
```

Comment:
So.. tensor means tensors for gradient and input / output and derivatives?

Reply:
yes. Tensors required to infer/train the model.